pkg/private/install.py.in

```diff
@@ -0,0 +1,194 @@
+                self._maybe_make_unowned_dir(os.path.join(relative_installdir, d))
+
+            logging.info("COPY_FROM_TREE %s <- %s", entry.dest, entry.src)
+            logging.info("CHMOD %s %s", entry.mode, entry.dest)
+            logging.info("CHOWN %s:%s %s", entry.user, entry.group, entry.dest)
+
+    def _install_symlink(self, entry):
+        raise NotImplementedError("symlinking not yet supported")
+        logging.info("SYMLINK %s <- %s", entry.dest, entry.link_to)
+        logging.info("CHMOD %s %s", entry.dest, entry.mode)
+        logging.info("CHOWN %s.%s %s", entry.dest, entry.user, entry.group)
+
+    def include_manifest_path(self, path):
+        with open(path, 'r') as fh:
+            self.include_manifest(fh)
+
+    def include_manifest(self, manifest_fh):
+        manifest_entries = json.load(manifest_fh)
+
+        for entry in manifest_entries:
+            # Swap out the source with the actual "runfile" location if we're
+            # called as a part of the build rather than "bazel run"
+            if not CALLED_FROM_BAZEL_RUN and entry[2] is not None:
+                entry[2] = os.path.join(RUNFILE_PREFIX, entry[2])
+            # Prepend the destdir path to all installation paths, if one is
+            # specified.
+            if self.destdir is not None:
+                entry[1] = os.path.join(self.destdir, entry[1])
+            entry_struct = manifest.ManifestEntry(*entry)
+            self.entries.append(entry_struct)
+
+    def do_the_thing(self):
+        for entry in self.entries:
+            if entry.entry_type == manifest.ENTRY_IS_FILE:
+                self._install_file(entry)
+            elif entry.entry_type == manifest.ENTRY_IS_LINK:
+                self._install_symlink(entry)
+            elif entry.entry_type == manifest.ENTRY_IS_DIR:
+                self._install_directory(entry)
+            elif entry.entry_type == manifest.ENTRY_IS_TREE:
+                self._install_treeartifact(entry)
+            else:
+                raise ValueError("Unrecognized entry type %d" % entry.entry_type)
+
+
+def main(args):
+    parser = argparse.ArgumentParser(
+        prog="bazel run -- ##TARGET_LABEL##",
+        description='Installer for bazel target ##TARGET_LABEL##',
+        fromfile_prefix_chars='@',
+    )
+    parser.add_argument('-v', '--verbose', action='count', default=0,
+                        help="Be verbose.  Specify multiple times to increase verbosity further")
+    parser.add_argument('-q', '--quiet', action='count', default=0,
+                        help="Be quieter.  Specify multiple times to be more quiet")
+    # TODO(nacl): consider supporting DESTDIR=/whatever syntax, like "make install"
+    parser.add_argument('--destdir', action='store', default=os.getenv("DESTDIR"),
+                        help="Installation root directory (defaults to DESTDIR environment variable)")
+
+    args = parser.parse_args()
+
+    loudness = args.verbose - args.quiet
```

Comment:
This seems baroquely fancy. Isn't the traditional enough?
default is WARNING
--quiet sets logging.ERROR and overrides verbose
--verbose goes up in level.

Reply:
I was trying to be traditional, but otherwise, yeah, you're right here.  Will adjust accordingly.